input/fsh/profiles/FiSchedulingAppointment.fsh

```diff
@@ -0,0 +1,121 @@
+* status ^requirements = "19 Ajanvarauksen tila"
+* status from $codeserver-booking-status-page (required)
+* status ^short = "suunniteltu | tilattu | varattu | peruttu | siirretty | alkanut | toteutunut | ehdotettu | saapumatta"
+* cancelationReason ^requirements = "19.1 Ajanvarauksen perumisen tai siirtämisen syy\r\n\r\nMandatory, if cancellation status is equivalent to \"Siirretty\" or \"Peruttu\""
+* cancelationReason.coding.system = "urn:oid:1.2.246.537.6.126.2008" (exactly)
+* cancelationReason.coding.system ^short = "THL - Palvelutapahtuman peruuntumisen tai siirtymisen syy"
+* serviceCategory ..1
+* serviceCategory ^requirements = "71.1 Palvelun luokka"
+* serviceCategory.coding.system = "urn:oid:1.2.246.537.6.88.2008" (exactly)
+* serviceCategory.coding.system ^short = "AR/YDIN - Palvelutapahtumaluokitus"
+* serviceCategory.coding.system ^definition = "hl7fi: koodilla ilmaistu tieto palveluluokasta, johon ajanvaraus kohdistuu Huom. Palveluluokka ilmaistaan THL - Sosiaali- ja terveyspalvelujen luokituksen avulla."
+* serviceType 1..1
+* serviceType ^requirements = "71 Palvelun nimi"
+* serviceType.coding.system = "urn:oid:1.2.246.537.6.49.201501" (exactly)
+* serviceType.coding.system ^short = "THL - Sosiaali- ja terveysalan palvelunimikkeistö"
+* serviceType.coding.system ^comment = "hl7fi: koodilla ilmaistu tieto palvelusta, johon ajanvaraus kohdistuu Huom. Palvelun nimi ilmaistaan THL - Sosiaali- ja terveysalan palvelunimikkeistön avulla.\r\n\r\nThe URI may be an OID (urn:oid:...) or a UUID (urn:uuid:...).  OIDs and UUIDs SHALL be references to the HL7 OID registry. Otherwise, the URI should come from HL7's list of FHIR defined special URIs or it should reference to some definition that establishes the system clearly and unambiguously."
+* serviceType.coding.system ^requirements = "71 Palvelun nimi"
+* specialty ..0
+* appointmentType ^requirements = "72 Asiointitapa"
+* appointmentType.coding.system = "1.2.246.537.6.884.2015" (exactly)
+* appointmentType.coding.system ^short = "hl7fi: asiointitapa"
+* appointmentType.coding.system ^definition = "hl7fi: koodilla ilmaistu tieto ammattihenkilön ja asiakkaan välisen asioinnin tavasta"
+* reasonCode ^slicing.discriminator.type = #exists
+* reasonCode ^slicing.discriminator.path = "coding"
+* reasonCode ^slicing.rules = #open
+* reasonCode contains
+    sliceSisaltotarkenne 0..* and
+    slicePalvelutarve 0..1
+* reasonCode[sliceSisaltotarkenne] ^definition = "hl7fi: 85 Varauksen sisältötarkenne, koodilla ilmaistu tarkentava tieto palvelun sisällöstä Huom. Tarkenteeksi voi valita useamman kuin yhden koodin, ja koodit voivat kuvata esimerkiksi sitä, mitä kuvantamis- tai laboratoriotutkimuksia tai toimenpiteitä suoritetaan."
```

Comment:
It's not clear to me that this would be a reason. To me, the `.serviceType` would be a much more logical place to express this kind of information.